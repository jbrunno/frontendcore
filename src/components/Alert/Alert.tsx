import React from 'react';
import { Alert as MuiAlert, AlertProps as MuiAlertProps } from '@mui/material';

export type AlertProps = MuiAlertProps;

export function Alert(props: AlertProps) {
  return <MuiAlert {...props} />;
}

export default Alert;
