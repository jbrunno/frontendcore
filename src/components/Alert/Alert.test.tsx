import React from 'react';
import { render } from '@test-utils/component-testing';
import Alert from './Alert';

test('renders the Alert', () => {
  const { container } = render(
    <Alert severity="error" variant="outlined" color="info">
      Alert test
    </Alert>,
  );
  expect(container).toMatchSnapshot();
});
