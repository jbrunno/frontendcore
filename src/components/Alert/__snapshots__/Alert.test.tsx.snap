// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Alert 1`] = `
.emotion-0 {
  background-color: #fff;
  color: #3B3B41DE;
  -webkit-transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  border-radius: 4px;
  box-shadow: none;
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.125rem;
  letter-spacing: 0.0094rem;
  font-family: Lato,sans-serif;
  background-color: transparent;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 6px 16px;
  color: rgb(40, 72, 98);
  border: 1px solid #64b6f7;
}

.emotion-0 .MuiAlert-icon {
  color: #64b6f7;
}

.emotion-1 {
  margin-right: 12px;
  padding: 7px 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 22px;
  opacity: 0.9;
}

.emotion-2 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: inherit;
}

.emotion-3 {
  padding: 8px 0;
  min-width: 0;
  overflow: auto;
}

<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-outlinedInfo MuiAlert-outlined emotion-0"
    role="alert"
  >
    <div
      class="MuiAlert-icon emotion-1"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit emotion-2"
        data-testid="ErrorOutlineIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
        />
      </svg>
    </div>
    <div
      class="MuiAlert-message emotion-3"
    >
      Alert test
    </div>
  </div>
</div>
`;
